File: lib/src/main/java/com/mobilabsolutions/payment/android/psdk/IntegrationInitialization.java
package com.mobilabsolutions.payment.android.psdk;

import com.mobilabsolutions.payment.android.psdk.internal.PaymentSdkComponent;
import com.mobilabsolutions.payment.android.psdk.internal.psphandler.Integration;

/**
 * @author <a href="ugi@mobilabsolutions.com">Ugi</a>
 */
public interface IntegrationInitialization {
    Integration initialize(PaymentSdkComponent paymentSdkComponent);
}

File: lib/src/main/java/com/mobilabsolutions/payment/android/psdk/internal/IntegrationInitialization.kt
/*
 * Copyright © MobiLab Solutions GmbH
 */

package com.mobilabsolutions.payment.android.psdk.internal

import com.mobilabsolutions.payment.android.psdk.PaymentMethodType
import com.mobilabsolutions.payment.android.psdk.internal.psphandler.Integration

/**
 * @author <a href="ugi@mobilabsolutions.com">Ugi</a>
 */
interface IntegrationInitialization {
    val enabledPaymentMethodTypes: Set<PaymentMethodType>

    fun initialize(stashComponent: StashComponent, url: String = ""): Integration

    fun initializedOrNull(): Integration?
}